### ISSUE TITLE
Strat writer life improvement: do not deprovision offers after failure

### ISSUE THREAD

**Issue by adhusson** [mangrovedao/mangrove#388]
Currently, offers that fail are fully deprovisioned. Part of the provision is consumed as penalty, the rest is returned to the taker.

For some strat writers (multiuser strats is a motivating example), it is hard to follow the accounting after a failed offer -- how much has been returned? how much was there originally?

In addition, there is a lack of uniformity -- successful offers don't get deprovisioned.

We propose changing the mechanism and let makers deprovision their failed offers themselves, if they want to. It is a little tricky to do right now because the provision is induced by the `gasprice`, `gasreq` and `offer_gasbase` of an offer. The main avenue being investigated is removing `gasprice` and `offer_gasbase` from the `offerDetail` struct, replacing it with a `provision` field, and editing that field every time the provision changes.

**Comment by espendk** [comment]
@adhusson This task has been completed, right?

**Comment by adhusson** [comment]
Task canceled for now.

Rationale: successful offers don't get deprovisioned because we know in advance they won't be penalized. However, a failing offer will have some bounty taken out of its provision. This bounty depends on the gas used during the offfer execution *and* during the offer posthook. Since the posthook has not been executed yet, we don't know by how much the provision will be penalized.

We don't have a satisfying solution for now so we are shelving this issue. Possible contenders for future updates are:
* Force the gas used in posthook not to go over some fixed global amount. Penalize offers by (gas used during execution)+(fixed global posthook gas stipend)
* Call makers a *third* time (after execute and posthook), with a fixed amount of gas, so they know by how much they were penalized
